web/modules/custom/joinup_seo/src/Plugin/simple_sitemap/UrlGenerator/JoinupUrlGeneratorBase.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\joinup_seo\Plugin\simple_sitemap\UrlGenerator;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Entity\Query\QueryInterface;
+use Drupal\Core\Language\LanguageManagerInterface;
+use Drupal\joinup_core\JoinupRelationManagerInterface;
+use Drupal\simple_sitemap\EntityHelper;
+use Drupal\simple_sitemap\Logger;
+use Drupal\simple_sitemap\Plugin\simple_sitemap\UrlGenerator\EntityUrlGenerator;
+use Drupal\simple_sitemap\Plugin\simple_sitemap\UrlGenerator\UrlGeneratorManager;
+use Drupal\simple_sitemap\Simplesitemap;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Base class for Joinup Url Generators.
+ *
+ * @package Drupal\joinup_seo\Plugin\simple_sitemap\UrlGenerator
+ */
+abstract class JoinupUrlGeneratorBase extends EntityUrlGenerator {
+
+  /**
+   * The url generator manager service.
+   *
+   * @var \Drupal\simple_sitemap\Plugin\simple_sitemap\UrlGenerator\UrlGeneratorManager
+   */
+  protected $urlGeneratorManager;
+
+  /**
+   * The Joinup relation manager service.
+   *
+   * @var \Drupal\joinup_core\JoinupRelationManagerInterface
+   */
+  protected $relationManager;
+
+  /**
+   * JoinupUrlGeneratorBase constructor.
```

Comment:
Does not conform to the Drupal documentation standards, it should be `Constructs a JoinupUrlGeneratorBase object.`.